run/local-cluster.go

```diff
@@ -0,0 +1,391 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Peter Mattis (peter.mattis@gmail.com)
+
+// Run using: go run local_cluster.go
+
+// +build ignore
+
+package main
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"strings"
+	"time"
+
+	"github.com/fsouza/go-dockerclient"
+)
+
+const (
+	dockerspyImage = "iverberk/docker-spy"
+	domain         = "local"
+)
+
+var cockroachImage = flag.String("i", "cockroachdb/cockroach-dev", "the docker image to run")
+var cockroachEntry = flag.String("e", "", "the entry point for the image")
+var numNodes = flag.Int("n", 3, "the number of nodes to start")
+var waitInterrupt = flag.Bool("w", false, "wait for an interrupt before existing")
+
+func prettyJSON(v interface{}) string {
+	pretty, err := json.MarshalIndent(v, "", "  ")
+	if err != nil {
+		panic(err)
+	}
+	return string(pretty)
+}
+
+func newDockerClient() *docker.Client {
+	client, err := docker.NewTLSClient(
+		os.ExpandEnv("${DOCKER_HOST}"),
+		os.ExpandEnv("${DOCKER_CERT_PATH}/cert.pem"),
+		os.ExpandEnv("${DOCKER_CERT_PATH}/key.pem"),
+		os.ExpandEnv("${DOCKER_CERT_PATH}/ca.pem"))
+	if err != nil {
+		log.Fatal(err)
+	}
+	return client
+}
+
+// remove a container (docker rm)
```

Comment:
The commenting is kind of sparse right now. Pay primary attention to the `main()` function and the `checkCluster`. My thought is that most of this file could be bundled into a package and be used for more involved acceptance tests. For example, it would naturally fall out of this to use the Go client to generate load. We can add, remove and pause machines. Etc.